src/atl/formula.rs

```diff
@@ -19,5 +20,76 @@ pub(crate) enum Phi {
         players: Vec<Player>,
         pre: Arc<Phi>,
         until: Arc<Phi>,
+    },
+}
+
+impl Display for Phi {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Phi::PROPOSITION(id) => f.write_fmt(format_args!("'{}'", id)),
+            Phi::NOT(formula) => {
+                f.write_str("¬")?;
+                formula.fmt(f)
+            }
+            Phi::OR(left, right) => {
+                left.fmt(f)?;
+                f.write_str(" ∨ ")?;
+                right.fmt(f)
+            }
+            Phi::NEXT { players, formula } => {
+                f.write_str("⟪")?;
+                f.write_str(
+                    players
+                        .iter()
+                        .map(|id| id.to_string())
+                        .collect::<Vec<String>>()
+                        .join(",")
+                        .as_str(),
+                )?;
+                f.write_str("⟫◯[")?;
+                formula.fmt(f)?;
+                f.write_str("]")
+            }
+            Phi::DESPITE_UNTIL {
+                players,
+                pre,
+                until,
+            } => {
+                f.write_str("⟪")?;
+                f.write_str(
+                    players
+                        .iter()
+                        .map(|id| id.to_string())
+                        .collect::<Vec<String>>()
+                        .join(",")
+                        .as_str(),
+                )?;
+                f.write_str("⟫(")?;
+                pre.fmt(f)?;
+                f.write_str(" 𝑼 ")?;
+                pre.fmt(f)?;
```

Comment:
```suggestion
                until.fmt(f)?;
```
Your graph revealed a bug lmao

Reply:
fixed in 214e2f7c860d00c78cb61a3f2e6eedebce554c81